src/librepgp/stream-sig.cpp

```diff
@@ -1273,3 +1274,73 @@ signature_check_subkey_revocation(pgp_signature_info_t *sinfo,
 
     return signature_check(sinfo, &hash);
 }
+
+void
+signature_list_destroy(list *sigs)
+{
+    for (list_item *li = list_front(*sigs); li; li = list_next(li)) {
+        free_signature((pgp_signature_t *) li);
+    }
+    list_destroy(sigs);
+}
+
+rnp_result_t
+process_pgp_signatures(pgp_source_t *src, list *sigs)
+{
+    bool             armored = false;
+    pgp_source_t     armorsrc = {0};
+    pgp_source_t *   origsrc = src;
+    pgp_signature_t *cursig = NULL;
+    rnp_result_t     ret = RNP_ERROR_GENERIC;
+
+    *sigs = NULL;
+    /* check whether signatures are armored */
+armoredpass:
```

Comment:
No, it would not be simple since jump to here will be done only if stream contains multiple armored messages, not a single one. It can be rewritten as loop however it will contain more nesting levels/more checks.